src/test/java/org/junit/tests/assertion/AssertionTest.java

```diff
@@ -674,4 +677,80 @@ public void assertNotEqualsIgnoresDeltaOnNaN() {
     public void assertNotEqualsIgnoresFloatDeltaOnNaN() {
         assertNotEquals(Float.NaN, Float.NaN, 1f);
     }
+
+    @Test(expected = AssertionError.class)
+    public void expectThrowsRequiresAnExceptionToBeThrown() {
+        expectThrows(nonThrowingRunnable());
+    }
+
+    @Test
+    public void expectThrowsIncludesNoMessageByDefault() {
+        try {
+            expectThrows(nonThrowingRunnable());
+            fail();
+        } catch (AssertionError ex) {
+            assertNull(ex.getMessage());
+        }
+    }
+
+    @Test
+    public void expectThrowsIncludesTheSuppliedMessage() {
+        try {
+            expectThrows("message", nonThrowingRunnable());
+            fail();
+        } catch (AssertionError ex) {
+            assertEquals("message", ex.getMessage());
+        }
+    }
+
+    @Test
+    public void expectThrowsReturnsTheSameObjectThrown() {
+        NullPointerException npe = new NullPointerException();
+
+        Throwable throwable = expectThrows(throwingRunnable(npe));
+
+        assertSame(npe, throwable);
+    }
+
+    @Test(expected = ClassCastException.class)
```

Comment:
Yes, I've already done that. You can throw an undeclared checked exception using a technique like Lombok uses to implement `sneakyThrow`, but wrapping in an `AssertionError` is the right thing to do if you want inline type assertions.